js/noms/src/list.js

```diff
@@ -118,20 +165,30 @@ export default class List<T: Value> extends Collection<IndexedSequence<any>> {
     return this.sequence.range(start, end);
   }
 
+  /**
+   * The number of elements in the list.
+   */
   get length(): number {
     return this.sequence.numLeaves;
   }
 }
 
+/**
+ * ListLeafSequence is used for the leaf lists of a list prolly-tree.
+ */
 export class ListLeafSequence<T: Value> extends IndexedSequence<T> {
   get chunks(): Array<Ref<any>> {
     return getValueChunks(this.items);
   }
 
   cumulativeNumberOfLeaves(idx: number): number {
+    // TODO: What does this do? Document me!
```

Comment:
Though to answer your question, this is an interface method which sequences implement, that returns how many leaf items there are up to an index within its sequence. For a leaf, this is trivially `idx + 1`.

For non-leaves it is more complex of course.